topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsConfiguration.java

```diff
@@ -4,246 +4,163 @@
-                                IllegalStateException.class, Status.FAILED_PRECONDITION,
-                                InvalidStateStoreException.class, Status.FAILED_PRECONDITION,
-                                Throwable.class, Status.INTERNAL
-                        )
-                );
-
-        Server server = ServerBuilder
-                .forPort(localHost.port())
-                .addService(
-                        ServerInterceptors.intercept(
-                                streamsStoreGrpcImpl,
-                                unknownStatusDescriptionInterceptor
-                        )
-                )
-                .addService(
-                        ServerInterceptors.intercept(
-                                streamsAsyncBiFunctionServiceGrpcImpl,
-                                unknownStatusDescriptionInterceptor
-                        )
-                )
-                .build();
-
-        return new Lifecycle() {
-            @Override
-            public void start() {
-                try {
-                    server.start();
-                } catch (IOException e) {
-                    throw new UncheckedIOException(e);
+    /* test */ TopicStore store;
+
+    public CompletionStage<TopicStore> start(Config config, Properties kafkaProperties) {
+        String storeTopic = config.get(Config.STORE_TOPIC);
+        String storeName = config.get(Config.STORE_NAME);
+
+        // check if entry topic has the right configuration
+        Admin admin = Admin.create(kafkaProperties);
+        return toCS(admin.describeCluster().nodes())
+                .thenApply(nodes -> new Context(nodes.size()))
+                .thenCompose(c -> toCS(admin.listTopics().names()).thenApply(c::setTopics))
+                .thenCompose(c -> {
+                    if (c.topics.contains(storeTopic)) {
+                        ConfigResource storeTopicConfigResource = new ConfigResource(ConfigResource.Type.TOPIC, storeTopic);
+                        return toCS(admin.describeTopics(Collections.singleton(storeTopic)).values().get(storeTopic))
+                            .thenApply(td -> c.setRf(td.partitions().stream().map(tp -> tp.replicas().size()).min(Integer::compare).orElseThrow()))
+                            .thenCompose(c2 -> toCS(admin.describeConfigs(Collections.singleton(storeTopicConfigResource)).values().get(storeTopicConfigResource))
+                                    .thenApply(cr -> c2.setMinISR(parseInt(cr.get(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG).value()))))
+                            .thenApply(c3 -> {
+                                if (c3.rf != Math.min(3, c3.clusterSize) || c3.minISR != c3.rf - 1) {
+                                    log.warn("Durability of the topic is not sufficient for production use - replicationFactor: {}, {}: {}",
+                                        c3.rf, TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, c3.minISR);
```

Comment:
I think the message should be more actionable: What do they need to do to make this warning do away? And there's no harm is mentioning the topic name explicitly.

Reply:
What would make it go away?
"Replication factor should be at least 3 and {} one less." ?

Reply:
Increase the replication factor of topic {} to at least 3 and configure the min.in.sync.replicas to {}